src/storage/v2/vertex_info_cache.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "storage/v2/vertex_info_cache.hpp"
+
+#include "storage/v2/property_value.hpp"
+
+namespace memgraph::storage {
+
+VertexInfoCache::VertexInfoCache() = default;
```

Comment:
General question regarding the cache...
Should it be thread safe or is it only being used inside a single-threaded execution session?

Reply:
Maybe in the future with parallel queries...for now AFAICT not needed